File: src/main/java/com/skyspace666/service/InFlightMealService.java
package com.skyspace666.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspace666.domain.InFlightMeal;
import com.skyspace666.dto.InFlightMealDTO;
import com.skyspace666.dto.InFlightMealSearchDTO;
import com.skyspace666.dto.InFlightMealPageDTO;
import com.skyspace666.dto.InFlightMealConvertCriteriaDTO;
import com.skyspace666.service.GenericService;
import com.skyspace666.dto.common.RequestDTO;
import com.skyspace666.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface InFlightMealService extends GenericService<InFlightMeal, Integer> {

	List<InFlightMeal> findAll();

	ResultDTO addInFlightMeal(InFlightMealDTO inFlightMealDTO, RequestDTO requestDTO);

	ResultDTO updateInFlightMeal(InFlightMealDTO inFlightMealDTO, RequestDTO requestDTO);

    Page<InFlightMeal> getAllInFlightMeals(Pageable pageable);

    Page<InFlightMeal> getAllInFlightMeals(Specification<InFlightMeal> spec, Pageable pageable);

	ResponseEntity<InFlightMealPageDTO> getInFlightMeals(InFlightMealSearchDTO inFlightMealSearchDTO);
	
	List<InFlightMealDTO> convertInFlightMealsToInFlightMealDTOs(List<InFlightMeal> inFlightMeals, InFlightMealConvertCriteriaDTO convertCriteria);

	InFlightMealDTO getInFlightMealDTOById(Integer inFlightMealId);







}

File: skyspace666_ui/src/services/InFlightMealService.js
import http from "../http-common"; 

class InFlightMealService {
  getAllInFlightMeals(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/inFlightMeal/inFlightMeals`, searchDTO);
  }

  get(inFlightMealId) {
    return this.getRequest(`/inFlightMeal/${inFlightMealId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/inFlightMeal?field=${matchData}`, null);
  }

  addInFlightMeal(data) {
    return http.post("/inFlightMeal/addInFlightMeal", data);
  }

  update(data) {
  	return http.post("/inFlightMeal/updateInFlightMeal", data);
  }
  
  uploadImage(data,inFlightMealId) {
  	return http.postForm("/inFlightMeal/uploadImage/"+inFlightMealId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new InFlightMealService();
